Packages/ClientRuntime/Sources/Serialization/SerializationUtils/TimestampSerdeUtils.swift

```diff
@@ -0,0 +1,274 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import func Foundation.floor
+
+/// Custom timestamp serialization formats
+/// https://awslabs.github.io/smithy/1.0/spec/core/protocol-traits.html#timestampformat-trait
+public enum TimestampFormat: CaseIterable {
+    /// Also known as Unix time, the number of seconds that have elapsed since 00:00:00 Coordinated Universal Time (UTC), Thursday, 1 January 1970, with optional fractional precision (for example, 1515531081.1234).
+    case epochSeconds
+    
+    /// Date time as defined by the date-time production in RFC3339 section 5.6 with no UTC offset and optional fractional precision (for example, 1985-04-12T23:20:50.52Z).
+    case dateTime
+    
+    /// An HTTP date as defined by the IMF-fixdate production in RFC 7231#section-7.1.1.1 (for example, Tue, 29 Apr 2014 18:30:38 GMT) with optional fractional seconds (for example, Sun, 02 Jan 2000 20:34:56.000 GMT).
+    case httpDate
+}
+
+/// A formatter that converts between dates and their smithy timestamp string representations.
+public struct TimestampFormatter {
+    /// The timestamp serialization format
+    public var format: TimestampFormat
```

Comment:
With the current implementation it doesn't **need** to be but imo a var is preferable to provide a better and more convenient API.  Since it's a struct, mutating the value doesn't have any downsides. Additionally, it stays consistent with the [DateFormatter](https://developer.apple.com/documentation/foundation/dateformatter) APIs such as `dateFormat` and `timeStyle`.

All that being said, I'm not opposed to making it a constant, so lmk.

Reply:
I always prefer immutability over mutability as default. `DateFormatter` that you linked doesn't seem Swifty enough rather look copy-paste from obj-c.

You can always go to `var` in future but can't to `let`. I would still prefer it a let.